pkg/virt-controller/watch/migration.go

```diff
@@ -830,6 +831,14 @@ func (c *MigrationController) sync(key string, migration *virtv1.VirtualMachineI
 				// will be marked as failed too.
 				return nil
 			}
+			if c.clusterConfig.NonRootEnabled() && util.CanBeNonRoot(vmi) == nil {
+				util.MarkAsNonroot(vmi)
```

Comment:
this VMI is from cache. a deep copy needs to be made somewhere.

Reply:
Thanks, I assumed we make a copy before passing to sync.

Reply:
Wait, why do we need to make an expensive copy here just to set one value on a k8s object?
Can't we just patch the VMI by name and tell it "set status.runtimeUser to 107"?

Reply:
Fixed. Let's see what CI will say.